app/src/main/java/framgia/com/ichat/screen/base/BaseActivity.java

```diff
@@ -0,0 +1,45 @@
+package framgia.com.ichat.screen.base;
+
+import android.app.ProgressDialog;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.widget.Toast;
+
+import framgia.com.ichat.R;
+
+public abstract class BaseActivity extends AppCompatActivity {
+    private ProgressDialog mProgressDialog;
+
+    protected abstract int getLayoutResource();
+
+    protected abstract void initComponents();
+
+    protected abstract void initData(Bundle savedInstanceState);
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(getLayoutResource());
+        initComponents();
+        initData(savedInstanceState);
+    }
+
+    public void showProgressDialog() {
+        if (mProgressDialog == null) {
+            mProgressDialog = new ProgressDialog(this);
+            mProgressDialog.setMessage(getString(R.string.progress_loading));
+            mProgressDialog.setIndeterminate(true);
+        }
+        mProgressDialog.show();
```

Comment:
nếu progress đang được show thì k cần gọi method này nữa
```suggestion
        if(!mProgressDialog.isShowing()) {
            mProgressDialog.show();
        }
```